### PR TITLE
#18285 parquet io docs

### LINKED ISSUE #18285: DOC: Parquet IO docs don't match code
#### Problem description

The [Parquet section of the IO docs](https://pandas-docs.github.io/pandas-docs-travis/io.html#parquet) mentions the following caveat:

> The format will NOT write an Index, or MultiIndex for the DataFrame and will raise an error if a non-default one is provided. You can simply .reset_index(drop=True) in order to store the index.

This conflicts with the [error messages](https://github.com/pandas-dev/pandas/blob/63e8527d32aaf6afe1cd4b2a7b3bfadb088c9a72/pandas/io/parquet.py#L149-L169), which recommends:

> you can .reset_index() to make the index into column(s)

#### Expected Output

AFAIK the error message is correct and the IO docs should remove `drop=True`:

> The format will NOT write an Index, or MultiIndex for the DataFrame and will raise an error if a non-default one is provided. You can simply .reset_index() in order to store the index.

I'm happy to submit a PR for this.

### PR DESCRIPTION
**Pull request by mattayes**
- [x] closes #18285 
- [x] tests ~added /~ passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] ~whatsnew entry~ (N/A)

### CODE REVIEW

File: doc/source/io.rst [hunk 1]
```diff
@@ -4492,7 +4492,7 @@ dtypes, including extension dtypes such as datetime with tz.
 Several caveats.
 
 - The format will NOT write an ``Index``, or ``MultiIndex`` for the ``DataFrame`` and will raise an
-  error if a non-default one is provided. You can simply ``.reset_index(drop=True)`` in order to store the index.
```
Comment (jorisvandenbossche) [inline doc/source/io.rst]: Can you add "or `reset_index(drop=True)` to ignore the index" ?

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks! Small comment

**Comment by mattayes** [comment]
@jorisvandenbossche Do you know what's up with those CodeCov report? I didn't touch any of the impacted files and the changes appear to be deleting and re-adding the same lines.

**Comment by jorisvandenbossche** [comment]
@mattayes you can just ignore that, codecov can be rather flaky ..

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
@mattayes Thanks for the quick PR!

**Comment by mattayes** [comment]
@jorisvandenbossche You're welcome, thanks for the quick review!

**Pull request merged**